extension/source/Confirm/Confirm.tsx

```diff
@@ -1,67 +1,124 @@
-import { ethers } from 'ethers';
-import { FunctionComponent, useEffect, useMemo, useState } from 'react';
-import { runtime } from 'webextension-polyfill';
-import TaskQueue from '../helpers/TaskQueue';
+import { FunctionComponent, useEffect, useState } from 'react';
+import { runtime, storage } from 'webextension-polyfill';
 
 // components, styles and UI
+import { Check, X, CaretLeft, CaretRight } from 'phosphor-react';
+import { ethers } from 'ethers';
 import Button from '../components/Button';
-import CompactQuillHeading from '../components/CompactQuillHeading';
-import { DEFAULT_CHAIN_ID_HEX } from '../env';
-import { useInputDecode } from '../hooks/useInputDecode';
-import formatCompactAddress from '../helpers/formatCompactAddress';
+import { TransactionStatus } from '../background/TransactionsController';
+import { SendTransactionParams } from '../types/Rpc';
+import TransactionCard from './TransactionCard';
+import onAction from '../helpers/onAction';
 
 const Confirm: FunctionComponent = () => {
-  const [id, setId] = useState<string>();
-  const [to, setTo] = useState<string>('0x');
-  const [value, setValue] = useState<string>('0');
-  const [data, setData] = useState<string>('0x');
-
-  // TODO (merge-ok) update component to work across multiple chains/networks
-  const chainId = DEFAULT_CHAIN_ID_HEX;
-  const { loading, method } = useInputDecode(data, to, chainId);
-
-  const cleanupTasks = useMemo(() => new TaskQueue(), []);
+  const [id, setId] = useState<string | null>();
+  const [actions, setActions] = useState<SendTransactionParams[]>([]);
+  const [current, setCurrent] = useState<number>(0);
 
   useEffect(() => {
     const params = new URL(window.location.href).searchParams;
-    setId(params.get('id') || '0');
-    setTo(params.get('to') || '0x');
-    setValue(params.get('value') || '0');
-    setData(params.get('data') || '0x');
+    const txid = params.get('id');
+    setId(txid);
+
+    const fetchTx = async () => {
+      const data = await storage.local.get('transactions');
+      const tx = data.transactions.value.outgoing.find(
+        (t: any) => t.id === txid,
```

Comment:
It had to be done this way because this popup is a new window and it does not have access to the internal RPC nor any other window objects.